koku/api/report/serializers.py

```diff
@@ -105,31 +135,62 @@ class BaseSerializer(serializers.Serializer):
     """A common serializer base for all of our serializers."""
 
     _opfields = None
+    _tagkey_support = None
 
     def __init__(self, *args, **kwargs):
-        """Initialize the FilterSerializer."""
-        tag_keys = kwargs.pop('tag_keys', None)
+        """Initialize the BaseSerializer."""
+        self.tag_keys = kwargs.pop('tag_keys', None)
         super().__init__(*args, **kwargs)
 
-        if tag_keys is not None:
-            tag_keys = {key: StringOrListField(child=serializers.CharField(),
-                                               required=False)
-                        for key in tag_keys}
-            # Add tag keys to allowable fields
-            self.fields.update(tag_keys)
+        if self.tag_keys is not None:
+            fkwargs = {'child': serializers.CharField(), 'required': False}
+            self._init_tag_keys(StringOrListField, fkwargs=fkwargs)
 
         if self._opfields:
             add_operator_specified_fields(self.fields, self._opfields)
 
     def validate(self, data):
-        """Validate incoming data."""
+        """Validate incoming data.
+
+        Args:
+            data    (Dict): data to be validated
+        Returns:
+            (Dict): Validated data
+        Raises:
+            (ValidationError): if field inputs are invalid
+        """
         handle_invalid_fields(self, data)
         return data
 
+    def _init_tag_keys(self, field, fargs=None, fkwargs=None):
```

Comment:
This method enables the Serializers to control which fields they use for initializing tag-enabled fields. 

This resolves @adberglund 's comment that OrderBySerializer should use ChoiceField.